source/extensions/common/wasm/context.cc

```diff
@@ -1457,6 +1457,11 @@ void Context::log(const Http::RequestHeaderMap* request_headers,
                   const Http::ResponseHeaderMap* response_headers,
                   const Http::ResponseTrailerMap* response_trailers,
                   const StreamInfo::StreamInfo& stream_info) {
+  // `log` may be called multiple times due to mid-request logging -- we only want to run on the
+  // last call.
+  if (!stream_info.requestComplete().has_value()) {
```

Comment:
I think we'll probably want Wasm access logs to conform, just not Wasm HTTP filters. Both share the code, so I imagine it's tricky to untangle. Maybe leave a comment or issue for later.

Reply:
Not sure what you mean -- i.e. the access logger should have the ability to write mid-connection logs, but the filter should not?

Reply:
Yeah, there are two different extension points - Wasm HTTP filters and Wasm access loggers. I guess it makes sense to make them both support periodic reporting at the same time, rather than just Wasm access loggers.